src/index.js

```diff
@@ -294,6 +294,31 @@ assign(Lrud.prototype, {
       this.setActiveChild(node.parent, id)
       this._bubbleActive(node.parent)
     }
+  },
+
+  /**
+   * insert a new node or replace a current node of the same id
+   * @func upsert
+   * @param {string} id
+   * @param {object} props
+   */
+  upsert: function (id, props) {
+    if (!id) throw new Error('Attempting to register with an invalid id')
+
+    var newNode = this._createNode(id, props)
+    var self = this
+
+    try {
```

Comment:
What's the reason for catching and rethrowing?

Reply:
Ah very good point, I had the try/catch there from when the catch was console logging out when I was testing. I'll remove the try/catch altogether.